ui/src/app/pages/workspace/workspace-edit.tsx

```diff
@@ -390,11 +390,13 @@ export const WorkspaceEdit = fp.flow(withCurrentWorkspace(), withCdrVersions(),
       const billingAccounts = (await userApi().listBillingAccounts()).billingAccounts;
 
       // Replace the free billing account with a new display name that has spend usage.
-      const displayFreeTierAccount: BillingAccount = {
-        ...billingAccounts.find(billingAccount => billingAccount.isFreeTier),
-        displayName: this.formatFreeTierBillingAccountName()
-      }
-      const displayBillingAccounts = billingAccounts.map(b => b.isFreeTier ? displayFreeTierAccount : b)
+      const displayBillingAccounts: Array<BillingAccount> = billingAccounts.map((b) => {
+        if (b.isFreeTier) {
+          return { ...b, displayName: this.formatFreeTierBillingAccountName()
+          };
```

Comment:
nit: this indentation is awkward, please either move this up to the previous line, or (probably preferred) space out the properties as you hadpreivoiusly:

```
{
  ...b,
  displayName: ...
};
```